ChildProject/pipelines/derivations.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import pandas as pd
+
+
+def acoustics():
+    pass
+
+def conversations(annotations: pd.DataFrame,
+                  interlocutors_1=('CHI',),
+                  interlocutors_2=('FEM', 'MAL', 'OCH'),
+                  max_interval=5000,
+                  min_delay=0):
+
+    """
+
+    :param annotations: dataframe of annotations
+    :type annotations: DataFrame
+    :param interlocutors_1: first group of interlocutors, default = ('CHI')
+    :type interlocutors_1: tuple
+    :param interlocutors_2: second group of interlocutors, default = ('FEM','MAL','OCH')
+    :type interlocutors_2: tuple
+    :param max_interval: maximum interval in ms for it to be considered a turn, default = 5000
+    :type max_interval: int
+    :param min_delay: minimum delay between somebody starting speaking
+    :type min_delay: int
+
+    :return: output annotation dataframe
+    :rtype: DataFrame
+    """
+
+    speakers = set(interlocutors_1 + interlocutors_2)
+
+    interactants = {k: set(interlocutors_2) for k in interlocutors_1}
+
+    for k in interlocutors_2:
+        if k in interactants:
+            interactants[k] = interactants[k] | set(interlocutors_1)
+        else:
+            interactants[k] = set(interlocutors_1)
+
+    annotations = annotations[annotations["speaker_type"].isin(speakers)].copy()
+
+    if annotations.shape[0]:
+
+        # store the duration between vocalizations
+        annotations["iti"] = annotations["segment_onset"] - annotations["segment_offset"].shift(1)
+        # store the previous speaker
+        annotations["prev_speaker_type"] = annotations["speaker_type"].shift(1)
+
+        annotations["delay"] = annotations["segment_onset"] - annotations["segment_onset"].shift(1)
+        annotations = annotations.reset_index(drop=True)
+
+        # not using absolute value for 'iti' is a choice and should be evaluated (we allow speakers to 'interrupt'
+        # themselves
```

Comment:
that's a cryptic comment! not sure I follow. Can you explain?

Reply:
I run out of time! but I should see the rest again

Reply:
The comment is about the fact that we check if ` interval < iti `  so the condition is met on turns that have negative iti . Which in practice means that the next speaker started talking before the first finished its turn. In order to change that behaviour, we now have the 'delay' parameter which allows to filter out vocalizations where the next vocalizations starts more than <delay> ms before the end of the first